library/src/main/kotlin/io/intrepid/bleidiom/BleTransformers.kt

```diff
@@ -45,7 +45,7 @@ internal fun fixCustomUUID(uuid: String) = when (uuid.length) {
 /**
  * Gets the [ByteArray] version of a [String] MAC-address.
  */
-val String.macAddress get() = ByteArray(6) { substring(it * 3, it * 3 + 2).toByte(16) }
+val String.macAddress get() = ByteArray(6) { substring(it * 3, it * 3 + 2).toShort(16).toPositiveInt().toByte() }
```

Comment:
`toPositiveInt()` is somewhat expensive since it can handle any sub-type of `Number`. Since you have more control here, do this instead.
`(substring(it * 3, it * 3 + 2).toInt(16) and 0x000000FF).toByte()`
Also, could you add a unit-test that tests this fix.